src/lib/mongoDriver/mongoConnectionPool.cpp

```diff
@@ -399,14 +399,20 @@ static std::string composeMongoUri
       optionPrefix = "&";
     }
 
-    if (timeout > 0)
+    // Check if timeout is not zero (which means it's either unset or explicitly set to a positive value)
+    if (timeout != 0)
     {
-      char buf[STRING_SIZE_FOR_LONG];
-      i2s(timeout, buf, sizeof(buf));
-      uri += optionPrefix + "connectTimeoutMS=" + buf;
-      optionPrefix = "&";
+        // If timeout is negative (e.g., -1 indicating it's unset), set it to the default value of 10000
+        if (timeout < 0)
+        {
+            timeout = 10000;
+        }
+
+        char buf[STRING_SIZE_FOR_LONG];
+        i2s(timeout, buf, sizeof(buf));
```

Comment:
Modifying an input parameter in the function use not to be a good idea... I'd suggest al anternative that doesn't modifies it:

```suggestion
        char buf[STRING_SIZE_FOR_LONG];
        if (timeout < 0)
        {
          i2s(10000, buf, sizeof(buf));
        }
        else
        {
          i2s(timeout, buf, sizeof(buf));
        }

```